ui/src/views/settings/modals/edit-user-modal/index.tsx

```diff
@@ -1,37 +1,60 @@
-import { Avatar, Button, Input, Label, Modal, Panel, Toolbar } from '@mergestat/blocks'
+import { useMutation } from '@apollo/client'
+import { Avatar, Button, HelpText, Input, Label, Modal, Panel, Toolbar } from '@mergestat/blocks'
 import { UserIcon, XIcon } from '@mergestat/icons'
 import cx from 'classnames'
 import React, { ChangeEvent, useCallback, useState } from 'react'
-import { useUserSettingsSetState } from 'src/state/contexts/user-settings.context'
-
-const roles = [
-  {
-    name: 'Admin',
-    desc: 'Admins have access to everything',
-  },
-  {
-    name: 'User',
-    desc: 'Users can create, read and update data, but can’t delete any data ',
-  },
-  {
-    name: 'Read-Only',
-    desc: 'Read-Only users can only read data',
-  },
-]
+import { UpdateUserRoleMutation } from 'src/api-logic/graphql/generated/schema'
+import { UPDATE_USER_PASSWORD, UPDATE_USER_ROLE } from 'src/api-logic/graphql/mutations/manage-users'
+import { useUserSettingsContext, useUserSettingsSetState } from 'src/state/contexts/user-settings.context'
+import { showSuccessAlert } from 'src/utils/alerts'
+import { TEST_IDS, USER_ROLES } from 'src/utils/constants'
 
 export const EditUserModal: React.FC = () => {
+  const [{ usernameEdit, roleEdit }] = useUserSettingsContext()
   const { setShowEditUserModal } = useUserSettingsSetState()
 
+  const [updatePassword] = useMutation(UPDATE_USER_PASSWORD, { errorPolicy: 'all' })
+
+  const [updateRole] = useMutation(UPDATE_USER_ROLE, {
+    errorPolicy: 'all',
+    onCompleted: (data: UpdateUserRoleMutation) => {
+      data.userMgmtSetUserRole && showSuccessAlert('User updated')
+    },
+    awaitRefetchQueries: true,
+    refetchQueries: () => ['getUsers']
+  })
+
   const close = useCallback(() => {
     setShowEditUserModal(false)
   }, [setShowEditUserModal])
 
-  const [Role, setRole] = useState('Admin')
+  const [role, setRole] = useState<string | null>(roleEdit)
+  const [password, setPassword] = useState<string>()
+  const [passwordConfirm, setPasswordConfirm] = useState<string>()
+  const [error, setError] = useState<boolean>(false)
 
   function onChangeValue(event: ChangeEvent<HTMLInputElement>) {
     setRole(event.target.value)
   }
 
+  const handleUpdateUser = () => {
+    setError(false)
+
+    // If password or confirm password are typed
+    if (password || passwordConfirm) {
+      if (password !== passwordConfirm) {
+        setError(true)
+      } else { // Update password and role as well
+        updatePassword({ variables: { username: usernameEdit, password } })
+        updateRole({ variables: { username: usernameEdit, role } })
```

Comment:
I think we should only call this `updateRole` if there is a role change - we should be able to detect this yeah? We can avoid the extra API request if only the password is being changed.